### PR TITLE
[DOC-373] Code location history

### PR DESCRIPTION
**Pull request by OwenKephart**
## Summary & Motivation

Creates a doc explaining how to view code location history.

## Changelog [New | Bug | Docs]

NOCHANGELOG

### CODE REVIEW

**Comment by OwenKephart** [comment]
* **#24002** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/24002?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @OwenKephart and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: docs/docs-beta/docs/dagster-plus/deployment/code-locations/code-location-history.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+2. In the row associated with the code location you're interested in, click **View history** on the **Updated** column.
+
+![Screenshot highlighting the "Updated" column for a code location)](/img/placeholder.svg)
+
+This will bring up a modal showing a history of every time that code location has been loaded, and metadata associated with that load.
+
+If you have connected Dagster+ to a GitHub or Gitlab repository, each row will have a link to the commit that was deployed at that point in time.
+
+If a code location has been deployed multiple times with identical metadata, these rows will be collapsed together. You can expand them by deselecting **Collapse similar entries** in the top left corner of the modal.
+
```
Comment (cmpadden) [inline docs/docs-beta/docs/dagster-plus/deployment/code-locations/code-location-history.md:24]:
```suggestion
```

File: docs/docs-beta/docs/dagster-plus/deployment/code-locations/code-location-history.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+## Rolling back to a previous code location version
+
+If you notice an issue with newly deployed code, or your code fails to deploy successfully, you can quickly roll back to a previously deployed image that's known to work properly.
+
+1. In the Dagster+ UI, navigate to the **Deployment** tab.
+2. In the row associated with the code location you're interested in, click **View history** on the **Updated** column.
+3. In the dropdown menu next to **View metadata**, select **Rollback to this version**.
```
Comment (cmpadden) [inline docs/docs-beta/docs/dagster-plus/deployment/code-locations/code-location-history.md:32]:
```suggestion
3. In the **Actions** column click the dropdown menu to the right of **View metadata**, select **Rollback to this version**.
```

I will say this two-in-one style of button is slightly unintuitive, at least when trying to describe it.

File: docs/docs-beta/docs/dagster-plus/deployment/code-locations/code-location-history.md (hunk 1 above)
Comment (cmpadden) [inline docs/docs-beta/docs/dagster-plus/deployment/code-locations/code-location-history.md:24]:
```suggestion
```

Empty line

**Review by cmpadden — COMMENTED** [review]
Nice and simple!

I was able to follow along in Purina.

File: docs/docs-beta/docs/dagster-plus/deployment/code-locations/code-location-history.md [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+![Screenshot highlighting the "Updated" column for a code location](/img/placeholder.svg)
```
Comment (cmpadden) [inline docs/docs-beta/docs/dagster-plus/deployment/code-locations/code-location-history.md:34]: Are there any "Next steps" we'd want to include at the end?

File: docs/docs-beta/docs/dagster-plus/deployment/code-locations/code-location-history.md [hunk 4]
```diff
@@ -0,0 +1,34 @@
+---
+title: "Code location history"
+displayed_sidebar: "dagsterPlus"
+sidebar_position: 4
+sidebar_label: "Code location history"
+---
+
```
[the remark this answers is not in the document]
Comment (erinkcochran87) [reply docs/docs-beta/docs/dagster-plus/deployment/code-locations/code-location-history.md:7]: ☝️ that

For prereqs, are there any Dagster+ permissions required to do anything in this guide?

File: docs/docs-beta/docs/dagster-plus/deployment/code-locations/code-location-history.md [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+# Code location history
+
+Dagster+ automatically tracks metadata every time a code location is loaded. This can be used to understand when changes have been made, and what those changes were. In addition, this metadata can be used to quickly redeploy an older version.
+
+## Viewing code location history
+
+1. In the Dagster+ UI, navigate to the **Deployment** tab.
+2. In the row associated with the code location you're interested in, click **View history** on the **Updated** column.
```
Comment (erinkcochran87) [inline docs/docs-beta/docs/dagster-plus/deployment/code-locations/code-location-history.md:15]:
```suggestion
2. In the row associated with the code location you're interested in, click **View history** in the **Updated** column.
```

Sorry for the nit - but it's inside the column

File: docs/docs-beta/docs/dagster-plus/deployment/code-locations/code-location-history.md [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+
+If a code location has been deployed multiple times with identical metadata, these rows will be collapsed together. You can expand them by deselecting **Collapse similar entries** in the top left corner of the modal.
+
+
+## Rolling back to a previous code location version
+
+If you notice an issue with newly deployed code, or your code fails to deploy successfully, you can quickly roll back to a previously deployed image that's known to work properly.
+
+1. In the Dagster+ UI, navigate to the **Deployment** tab.
+2. In the row associated with the code location you're interested in, click **View history** on the **Updated** column.
```
Comment (erinkcochran87) [inline docs/docs-beta/docs/dagster-plus/deployment/code-locations/code-location-history.md:31]:
```suggestion
2. In the row associated with the code location you're interested in, click **View history** in the **Updated** column.
```

Same here

File: docs/docs-beta/docs/dagster-plus/deployment/code-locations/code-location-history.md (hunk 3 above)
Comment (erinkcochran87) [inline docs/docs-beta/docs/dagster-plus/deployment/code-locations/code-location-history.md:34]: Also, does anything happen after the user selects this? Is there confirmation in the UI or something?

File: docs/docs-beta/docs/dagster-plus/deployment/code-locations/code-location-history.md [hunk 7]
```diff
@@ -0,0 +1,34 @@
+---
+title: "Code location history"
```
Comment (erinkcochran87) [inline docs/docs-beta/docs/dagster-plus/deployment/code-locations/code-location-history.md:2]: This doesn't need to be merge-blocking, but I think we could get a more actionable/useful title here.

Ideally a user would be able to tell from the sidebar and title that this is where they'd go to rollback if they have a problem. wdyt, @cmpadden? I'm struggling to come up with something that isn't a mouthful.

Comment (erinkcochran87) [reply docs/docs-beta/docs/dagster-plus/deployment/code-locations/code-location-history.md:2]: But in that case, what does `deployment` refer to? Rolling back an entire full deployment? We use that term in a number of ways, unfortunately :|

Comment (cmpadden) [reply docs/docs-beta/docs/dagster-plus/deployment/code-locations/code-location-history.md:2]: Good point.

"image rollback", "version rollback", "Code location history and image rollback"

You're right that they're a mouthful, hah.

File: docs/docs-beta/docs/dagster-plus/deployment/code-locations/code-location-history.md (hunk 3 above)
Comment (OwenKephart) [reply docs/docs-beta/docs/dagster-plus/deployment/code-locations/code-location-history.md:34]: There is no confirmation dialog here, it just goes right ahead and does it

File: docs/docs-beta/docs/dagster-plus/deployment/code-locations/code-location-history.md (hunk 7 above)
Comment (OwenKephart) [reply docs/docs-beta/docs/dagster-plus/deployment/code-locations/code-location-history.md:2]: I updated to "Code location history and rollbacks" -- it's a bit vague but at least it gives people who explicitly are looking for rollbacks a breadcrumb

File: docs/docs-beta/docs/dagster-plus/deployment/code-locations/code-location-history.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (OwenKephart) [reply docs/docs-beta/docs/dagster-plus/deployment/code-locations/code-location-history.md:7]: you need editor permissions for rollbacks, so I'll add a note in that section

[the remark this answers is not in the document]
Comment (OwenKephart) [reply docs/docs-beta/docs/dagster-plus/deployment/code-locations/code-location-history.md:7]: @cmpadden Struggling to think of what to put in the "what you'll learn" section that isn't just a duplicate of the two section titles haha -- any thoughts there?

File: docs/docs-beta/docs/dagster-plus/deployment/code-locations/code-location-history.md (hunk 3 above)
Comment (OwenKephart) [reply docs/docs-beta/docs/dagster-plus/deployment/code-locations/code-location-history.md]: added!

File: docs/docs-beta/docs/dagster-plus/deployment/code-locations/code-location-history.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (cmpadden) [reply docs/docs-beta/docs/dagster-plus/deployment/code-locations/code-location-history.md:7]: I think it's okay to be a little repetitive, I've added a basic "what you'll learn" section

**Review by cmpadden — APPROVED** [review]
😗

**Pull request merged**